core/new-gui/src/app/workspace/component/codearea-custom-template/codearea-custom-template.component.ts

```diff
@@ -20,20 +20,36 @@ import { CoeditorPresenceService } from "../../service/workflow-graph/model/coed
   templateUrl: "./codearea-custom-template.component.html",
   styleUrls: ["./codearea-custom-template.component.scss"],
 })
-export class CodeareaCustomTemplateComponent extends FieldType {
+export class CodeareaCustomTemplateComponent extends FieldType<any> implements AfterViewInit {
   dialogRef: MatDialogRef<CodeEditorDialogComponent> | undefined;
+  disabled: boolean = false;
 
   constructor(public dialog: MatDialog, private coeditorPresenceService: CoeditorPresenceService) {
     super();
     this.handleShadowingMode();
   }
 
+  ngAfterViewInit() {
+    this.handleDisabled();
+  }
+
+  /**
+   * Syncs the disabled status of the button with formControl.
+   * Used to fit the unit test since undefined might occur.
+   */
+  handleDisabled(): void {
+    if (this.field !== undefined)
+      this.field.formControl.statusChanges.pipe(untilDestroyed(this)).subscribe(() => {
```

Comment:
I again think this should listen to WorkflowActionService, rather than formControl's status change.

Reply:
ok, understood. so unlock for logic change is done locally, without going through `WorkflowActionService`. I have a feeling that it's better to manage those disabled states within `WorkflowActionService`, but I do not have a better suggestion for now. Can you leave comments/todo in code so we aware of this behavior, and to revisit in the future?

Reply:
Added.